zkevm-circuits/src/evm_circuit/op_execution.rs

```diff
@@ -213,6 +215,22 @@ pub(crate) struct OpExecutionGadget<F> {
     push_gadget: PushGadget<F>,
     lt_gadget: LtGadget<F>,
     pop_gadget: PopGadget<F>,
+    dup1_gadget: DupGadget<F, 0x80>,
+    dup2_gadget: DupGadget<F, 0x81>,
+    dup3_gadget: DupGadget<F, 0x82>,
+    dup4_gadget: DupGadget<F, 0x83>,
```

Comment:
have used const generic to wrap all dupx code into dup.rs gadget, but at the call side, we also have to distinguish them one by one after have tried vec! & encapsulations.  if you have good idea about this, welcome to point out . thanks

Reply:
change index_offset to expression make sense here and applied